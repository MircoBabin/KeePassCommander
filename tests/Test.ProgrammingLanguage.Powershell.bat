@echo off
    setlocal
    call "%~dp0Utils\Utils.test-setup.bat" "%~f0"
    
    powershell -NoProfile -ExecutionPolicy Bypass -file "%~dp0Test.ProgrammingLanguage.Powershell.ps1" "%KeePassCommandExe%"> "%KeePassCommand_TestOutput%"
    
    call "%~dp0Utils\Utils.diff.bat"
    if errorlevel 1 exit /b 1
    exit /b 0