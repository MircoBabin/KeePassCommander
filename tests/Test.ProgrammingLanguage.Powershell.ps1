param(
    [Parameter(Mandatory=$true)][string]$KeePassCommand_exe=''
)
    
$KeePassEntry_ps1 = Join-Path -Path $PSScriptRoot  -ChildPath '..\src\ProgrammingLanguagesConnectors\KeePassEntry.ps1'

# include KeePassEntry.ps1 containing function KeePassEntry
. $KeePassEntry_ps1

$options = [pscustomobject]@{
    KeePassCommandExe = $KeePassCommand_exe
    FieldNames = @('extra field 1', 'extra password 1')
    AttachmentNames = @('example_attachment.txt')
}

$object = KeePassEntry -title "Sample Entry" -options $options
    
$object | ConvertTo-Json -Compress | Out-String | % Trim | Write-Host -NoNewline
