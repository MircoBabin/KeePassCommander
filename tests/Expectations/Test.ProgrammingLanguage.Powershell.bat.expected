{"title":"Sample Entry","username":"User Name","password":"Password","url":"https://keepass.info/","urlscheme":"https","urlhost":"keepass.info","urlport":"443","urlpath":"/","notes":"Notes Één €ûrò","fields":[{"name":"extra field 1","value":"Some extra data"},{"name":"extra password 1","value":"Some extra password"}],"attachments":[{"name":"example_attachment.txt","value":"84 104 105 115 32 105 115 32 97 110 32 97 116 116 97 99 104 109 101 110 116 46 13 10 13 10 87 105 116 104 32 116 104 114 101 101 32 108 105 110 101 115 32 105 110 32 105 116 46"}]}

